DatadogSessionReplay/Sources/Utilities/CGRectExtensions.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+internal extension CGRect {
+    enum HorizontalAlignment {
+        case left, right, center
+    }
+
+    enum VerticalAlignment {
+        case top, bottom, middle
+    }
+
+    /// Puts this rect inside other rect with given alignment.
+    func putInside(
```

Comment:
This might be better called `alignWith`, since when aligned left or right this rectangle actually ends up outside the bounds of the rect it will be "inside".

Reply:
Nice ascii art @ncreated 🎨

Reply:
Ah, you're right! For some reason I thought `.origin` was the center, not the bottom left corner.